src/features/import/hooks/useUIDataColumn.ts

```diff
@@ -30,7 +30,16 @@ export default function useUIDataColumn(
   const cellValues = rows.map((row) => row.data[columnIndex]);
   const numRowsByUniqueValue = new Map<string | number, number>();
 
-  cellValues.forEach((value, idx) => {
+  let cellValuesSorted = [];
+  if (firstRowIsHeaders) {
+    const headerItem = cellValues[0];
+    const values = cellValues.slice(1);
+    cellValuesSorted = [headerItem, ...values.sort()];
+  } else {
+    cellValuesSorted = cellValues.sort();
+  }
```

Comment:
At first I was a little bit uncomfortable making this change in `useUIDataColumn()`, instead of doing it in the UI component. But as I was investigating, I realized that the way this hook is designed the array needs to be sorted in here.

I think this is a flaw in the way this hook has been designed, and not a flaw in how you've chosen to solve this problem.